pkg/api/v1alpha1/tinkerbellmachineconfig.go

```diff
@@ -0,0 +1,197 @@
+		Version:       "0.1",
+		Name:          name,
+		GlobalTimeout: 6000,
+		Tasks: []workflow.Task{{
+			Name:       name,
+			WorkerAddr: "{{.device_1}}",
+			Volumes: []string{
+				"/dev:/dev",
+				"/dev/console:/dev/console",
+				"/lib/firmware:/lib/firmware:ro",
+			},
+			Actions: []workflow.Action{
+				{
+					Name:    "stream-image",
+					Image:   "image2disk:v1.0.0",
+					Timeout: 360,
+					Environment: map[string]string{
+						"IMG_URL":    "http://10.62.2.49:8080/ubuntu-2004-kube-v1.21.5.gz",
+						"DEST_DISK":  "/dev/sda",
+						"COMPRESSED": "true",
+					},
+				},
+				{
+					Name:    "install-openssl",
+					Image:   "cexec:v1.0.0",
+					Timeout: 90,
+					Environment: map[string]string{
+						"BLOCK_DEVICE":        "/dev/sda1",
+						"FS_TYPE":             "ext4",
+						"CHROOT":              "y",
+						"DEFAULT_INTERPRETER": "/bin/sh -c",
+						"CMD_LINE":            "apt -y update && apt -y install openssl",
+					},
+				},
+				{
+					Name:    "write-netplan",
+					Image:   "writefile:v1.0.0",
+					Timeout: 90,
+					Environment: map[string]string{
+						"DEST_DISK": "/dev/sda1",
+						"FS_TYPE":   "ext4",
+						"DEST_PATH": "/etc/netplan/config.yaml",
+						"CONTENTS": `network:
+  version: 2
+  renderer: networkd
+  ethernets:
+    eno1:
+        dhcp4: true
+    eno2:
+        dhcp4: true
+    eno3:
+        dhcp4: true
+    eno4:
+        dhcp4: true`,
+						"UID":     "0",
+						"GID":     "0",
+						"MODE":    "0644",
+						"DIRMODE": "0755",
+					},
+				},
+				{
+					Name:    "add-tink-cloud-init-config",
+					Image:   "writefile:v1.0.0",
+					Timeout: 90,
+					Environment: map[string]string{
+						"DEST_DISK": "/dev/sda1",
+						"FS_TYPE":   "ext4",
+						"DEST_PATH": "/etc/cloud/cloud.cfg.d/10_tinkerbell.cfg",
+						"CONTENTS": `datasource:
+  Ec2:
+    metadata_urls: ["http://10.62.2.49:50061"]
+    strict_id: false
+system_info:
+  default_user:
+    name: tink
+    groups: [wheel, adm]
+    sudo: ["ALL=(ALL) NOPASSWD:ALL"]
+    shell: /bin/bash
+manage_etc_hosts: localhost
+warnings:
+  dsid_missing_source: off`,
+						"UID":     "0",
+						"GID":     "0",
+						"MODE":    "0600",
+						"DIRMODE": "0700",
+					},
+				},
+				{
+					Name:    "add-tink-cloud-init-ds-config",
+					Image:   "writefile:v1.0.0",
+					Timeout: 90,
+					Environment: map[string]string{
+						"DEST_DISK": "/dev/sda1",
+						"FS_TYPE":   "ext4",
+						"DEST_PATH": "/etc/cloud/ds-identify.cfg",
+						"CONTENTS":  `datasource: Ec2`,
+						"UID":       "0",
+						"GID":       "0",
+						"MODE":      "0600",
+						"DIRMODE":   "0700",
+					},
+				},
+				{
+					Name:    "kexec-image",
+					Image:   "kexec:v1.0.0",
+					Timeout: 90,
+					Pid:     "host",
+					Environment: map[string]string{
+						"BLOCK_DEVICE": "/dev/sda1",
+						"FS_TYPE":      "ext4",
+					},
+				},
+			},
+		}},
+	}
+}
+
+func (c *TinkerbellMachineConfigGenerate) APIVersion() string {
```

Comment:
Do you think we need this? `config.TypeMeta.APIVersion` isn't particularly verbose. Ditto for the other accessor methods.